KsApi/combine/CombineTestObserver.swift

```diff
@@ -27,4 +33,161 @@ public final class CombineTestObserver<Value, Error: Swift.Error> {
     }
     .store(in: &self.subscriptions)
   }
+
+  /// Get all of the next values emitted by the signal.
+  public var values: [Value] {
+    var values: [Value] = []
+    for event in self.events {
+      switch event {
+      case let .value(v):
+        values.append(v)
+      default: break
+        // do nothing
+      }
+    }
+
+    return values
+  }
+
+  /// Get the last value emitted by the signal.
+  public var lastValue: Value? {
+    return self.values.last
+  }
+
+  /// `true` if at least one `.Next` value has been emitted.
+  public var didEmitValue: Bool {
+    return self.values.count > 0
+  }
+
+  /// The failed error if the signal has failed.
+  public var failedError: Error? {
+    var errors: [Error] = []
+    for event in self.events {
+      switch event {
+      case let .error(e):
+        errors.append(e)
+      default: break
+        // do nothing
+      }
+    }
+
+    assert(
+      errors.count <= 1,
+      "I'm pretty sure a Combine publisher can only ever emit one error. If this fails, we've learned something new today."
```

Comment:
Nit: Do we want this check in our codebase? I'm okay with it since I'm kind of curious, but it feels a bit unprofessional/informal.

Reply:
Ha, I like throwing a little informality into code sometimes, but if you want me to make the language more neutral I definitely can.

Reply:
I'm used to an "avoid we" policy and I do have a slight preference for more neutral language, but I also think this is kind of fun and it's still clear who "we" is. So up to you.

Reply:
Interesting point on "we"! That's one I hadn't heard before.